src/simulators.ts

```diff
@@ -98,7 +98,21 @@ export async function boot(udid: string): Promise<void>
 {
     logger.log(`Booting simulator (udid: ${udid}) if required`);
     let time = new Date().getTime();
-    await _execFile('xcrun', ['simctl', 'bootstatus', udid, '-b']);
+
+    let bootPromise = _execFile('xcrun', ['simctl', 'bootstatus', udid, '-b']);
+
+    // Wait for boot or 1s and fire open command to launch the Simulator app
+    // The delay is required as without the delay there is some race condition
+    // and it throws an error saying can't boot already booted simulator
+    Promise.race([
+            bootPromise,
+            new Promise(resolve => setTimeout(resolve, 1_000))
+        ])
+        .then(() => _execFile('open', ['-g', '-a', 'Simulator', '--args', '-CurrentDeviceUDID', udid]));
+
+    // Wait for boot to complete
+    await bootPromise;
+
```

Comment:
Did some digging here... The problem is that when Simulator.app opens it tries to boot the last simulator by default (using `-CurrentDeviceUDID` just tells it what to use as the last one, turns out the args are just a way to set one-time preferences, see `defaults read com.apple.iphonesimulator CurrentDeviceUDID`). To fix this, one can change the Simulator setting by passing `-StartLastDeviceOnLaunch 0`, which turns off the setting `When Simulator starts boot the most recently used simulator` just for the single run. I don't think you even need to set `CurrentDeviceUDID`.

```suggestion
await _execFile('open', ['-g', '-a', 'Simulator', '--args', '-StartLastDeviceOnLaunch', '0']);
await _execFile('xcrun', ['simctl', 'bootstatus', udid, '-b']);
```

Reply:
How do you know about these params? I was tring to find other possible params to see if something was interesting.

Reply:
A series of realizations lol. I saw the option `When Simulator starts boot the most recently used simulator` in Simulator preferences and was wondering if we could turn it off programmatically for a single run (without doing a `defaults write` call)... then run `defaults read com.apple.iphonesimulator` and realized `CurrentDeviceUDID` was just a setting, so I unchecked the other option and saw in the `defaults` output that it was called `StartLastDeviceOnLaunch`. Finally I tried passing it as an arg and it worked... which seems to indicate the args to Simulator.app allow to set settings values for the single run.